packages/application/src/frontend.ts

```diff
@@ -349,4 +352,21 @@ namespace Private {
    * ersatz command.
    */
   export const CONTEXT_MENU_INFO = '__internal:context-menu-info';
+
+  /**
+   * Returns whether the element is itself, or a child of, an element with the `jp-suppress-context-menu` data attribute.
+   */
+  export function suppressContextMenu({
+    dataset,
+    parentElement
+  }: HTMLElement): boolean {
+    // Any value of data property means its active.
+    if (typeof dataset.jpSuppressContextMenu === 'string') {
+      return true;
+    }
+    if (parentElement) {
+      return suppressContextMenu(parentElement);
```

Comment:
Hooray for [tail-call optimization](https://bugzilla.mozilla.org/show_bug.cgi?id=723959) in ES6!

Reply:
Wow, I didn't know that! Too bad firefox doesn't implement it yet

Reply:
Another way to do it is to just do a while or do-while loop instead of using recursion, like in https://github.com/jupyterlab/jupyterlab/blob/6e0b32e4b86d950dd2a879b7a80a57c66481b28c/packages/application/src/frontend.ts#L126-L131

Reply:
Yes, and this approach is preferred due to recursion perf overhead.

Reply:
Sounds good, I'll change that. Thanks for the pointer

Reply:
Also note that you can use `.parentElement`, and I think not need to do the `node !== node.parentNode` test. I think something like this should work:

```js
while (node) {
    if (typeof node.dataset.jpSuppressContextMenu === 'string') {
      return true;
    }
    node = node.parentElement;
}
return false;
```

Reply:
Fixed! Removed three lines too so double win